src/macaron/dependency_analyzer/cyclonedx.py

```diff
@@ -35,155 +46,633 @@ def deserialize_bom_json(file_path: Path) -> dict:
+
+        Returns
+        -------
+        dict[str, DependencyInfo]
+            A dictionary where artifacts are grouped based on ``artifactId:groupId``.
+        """
+        deps_resolved: dict[str, DependencyInfo] = {}
+
+        build_tools = main_ctx.dynamic_data["build_spec"]["tools"]
+        if not build_tools:
+            # If the repository is not available, we use the first build tool object that has
+            # matched the software component's PackageURL type to allow a customized dependency resolution logic.
+            if main_ctx.dynamic_data["build_spec"]["purl_tools"]:
+                build_tools = [main_ctx.dynamic_data["build_spec"]["purl_tools"][0]]
+
+            # Check if we cannot still find a matching build tool.
+            if not build_tools:
+                logger.info("Unable to find any valid build tools.")
+                return {}
+
+        # Grab dependencies for each build tool, collate all into the deps_resolved.
+        for build_tool in build_tools:
+            try:
+                # We allow dependency analysis if SBOM is provided but no repository is found.
+                dep_analyzer = build_tool.get_dep_analyzer()
+            except DependencyAnalyzerError as error:
+                logger.error("Unable to find a dependency analyzer for %s: %s", build_tool.name, error)
+                return {}
+
+            if isinstance(dep_analyzer, NoneDependencyAnalyzer):
+                logger.info(
+                    "Dependency analyzer is not available for %s",
+                    build_tool.name,
+                )
+                continue
 
-def get_root_component(root_bom_path: Path) -> dict | None:
-    """Get the root dependency component.
+            if sbom_path:
+                logger.info("Getting the dependencies from the SBOM defined at %s.", sbom_path)
 
-    Parameters
-    ----------
-    root_bom_path : str
-        Path to the root bom.json file.
+                deps_resolved = dep_analyzer.get_deps_from_sbom(sbom_path, main_ctx.component)
 
-    Returns
-    -------
-    dict | None
-        The root component.
-    """
-    try:
-        root_bom = deserialize_bom_json(root_bom_path)
-    except CycloneDXParserError as error:
-        logger.error(error)
-        return None
-    try:
-        return root_bom.get("metadata").get("component")  # type: ignore
-    except AttributeError as error:
-        logger.error(error)
+                # Use repo finder to find more repositories to analyze.
+                if defaults.getboolean("repofinder", "find_repos"):
+                    DependencyAnalyzer._resolve_more_dependencies(deps_resolved)
 
-    return None
+                return deps_resolved
 
+            if not main_ctx.component.repository:
+                logger.info(
+                    "Unable to find a repository and no SBOM is provided as input. Analyzing the dependencies will be skipped."
+                )
+                return {}
```

Comment:
@tromai FYI: I fixed a bug here when neither a repository nor SBOM is available.

Reply:
Hmmm it might not be relevant to this PR. This block operates somewhat irrelevant to the build tool detected for the software component and can be moved outside of the for loop too
https://github.com/oracle/macaron/blob/569b511ae5a1a22d74b0c2a185643c5ba79ced50/src/macaron/dependency_analyzer/cyclonedx.py#L397-L412

Reply:
The `dep_analyzer` is obtained from the build tool though: https://github.com/oracle/macaron/blob/569b511ae5a1a22d74b0c2a185643c5ba79ced50/src/macaron/dependency_analyzer/cyclonedx.py#L385